AK/DOSPackedTime.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2020, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/DOSPackedTime.h>
+
+namespace AK {
+
+Time time_from_packed_dos(DOSPackedDate date, DOSPackedTime time)
+{
+    if (date.value == 0)
+        return Time();
+
+    return Time::from_timestamp(1980 + date.year, date.month, date.day, time.hour, time.minute, time.second * 2, 0);
+}
+
+DOSPackedDate to_packed_dos_date(unsigned year, unsigned month, unsigned day)
+{
+    DOSPackedDate date;
+    date.value = (year - 1980) << 9 | month << 5 | day;
+
+    return date;
+}
+
+DOSPackedTime to_packed_dos_time(unsigned hour, unsigned minute, unsigned second)
+{
+    DOSPackedTime time;
+    time.value = hour << 11 | minute << 5 | second >> 1;
```

Comment:
Can we write to the named bitfields (year, month, date, hour minute, second) rather than writing to the value field and shifting the bits into place? It would be a lot easier to read and I trust the compiler to coalesce the writes.